lib/engine/game/g_rolling_stock/game.rb

```diff
@@ -188,6 +247,57 @@ def update_offering(company)
           @offering << next_to_offer
           @on_deck << next_to_offer
         end
+
+        def share_prices
+          PAR_PRICES.values.flatten.uniq.map { |p| lookup_price(p) }
+        end
+
+        def ipo_companies
+          @companies.select { |c| c.owner&.player? && c.owner != @foreign_investor }.sort_by(&:value)
+        end
+
+        def available_par_prices(company)
+          PAR_PRICES[@company_stars[company.sym]].map { |p| lookup_price(p) }.select { |p| p.corporations.empty? }
+        end
+
+        def lookup_price(price)
```

Comment:
this looks really inefficient, you're doing n^2 operations every time you  do lookup_price?

Reply:
Good point. Thanks.